### PR TITLE
Unify predicate for outbound tasks

### PR DESCRIPTION
**Pull request by bergundy**
## What changed?

Add a single predicate for outbound tasks consisting of "task group", "namespace ID", and "destination" to replace three distinct predicates under an "AND" predicate.

Also noticed we were missing OR and AND support for `OutboundTaskGroupPredicate` so I added that for completeness.

## Why?

With the limited merging capabilities of the multi-cursor algorithm today, we've seen the previous approach generate an unreasonable amount of predicate objects. This could cause persistence issues with all the predicates stores in the single shard info DB record as well as cause increased memory usage.

## How did you test it?

Added unit tests.

### CODE REVIEW

File: service/history/tasks/predicates.go [hunk 1]
```diff
@@ -174,6 +174,52 @@ func (t *TypePredicate) Equals(predicate Predicate) bool {
 	return maps.Equal(t.Types, typePrediate.Types)
 }
 
+// OutboundTaskGroupNamespaceIDAndDestination is the key for grouping tasks by task type namespace ID and destination.
+type OutboundTaskGroupNamespaceIDAndDestination struct {
```
Comment (yycptt) [inline service/history/tasks/predicates.go:178]: nit: guess we don't really need "Outbound" in the name

File: proto/internal/temporal/server/api/persistence/v1/predicates.proto [hunk 2]
```diff
@@ -77,3 +78,11 @@ message OutboundTaskGroupPredicateAttributes {
     repeated string groups = 1;
 }
 
+message OutboundTaskPredicateAttributes {
+    message Group {
+        string task_group = 1;
```
Comment (yycptt) [inline proto/internal/temporal/server/api/persistence/v1/predicates.proto:83]: It's kinda weird it's called group but can only be one type of outbound task.

Shall we make this repeated string (will certainly make and/or logic more complex)? or just call it state_ machine_task_type?

**Review by yycptt — APPROVED** [review]

File: proto/internal/temporal/server/api/persistence/v1/predicates.proto (hunk 2 above)
Comment (bergundy) [reply proto/internal/temporal/server/api/persistence/v1/predicates.proto:83]: It's not state machine task type, it's a grouping of tasks by type.
I intend in the future to support grouping start and cancel tasks together for example.

File: service/history/tasks/predicates.go (hunk 1 above)
Comment (bergundy) [reply service/history/tasks/predicates.go:178]: Sure, I can change that.

Comment (bergundy) [reply service/history/tasks/predicates.go]: Done.

**Pull request merged**